clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -680,6 +694,7 @@ public NetworkClientDelegate.UnsentRequest toUnsentRequest() {
             }
             if (memberInfo.memberEpoch.isPresent()) {
                 data = data.setGenerationIdOrMemberEpoch(memberInfo.memberEpoch.get());
+                lastEpochSentOnCommit = Optional.of(memberInfo.memberEpoch.get());
```

Comment:
It seems we can assign `memberInfo.memberEpoch` to `lastEpochSentOnCommit` as `Optional` is a POJO

Reply:
For another, should we reset `lastEpochSentOnCommit` to none if `memberInfo.memberEpoch` is not defined?

Reply:
Both done, great catch the 2nd one! I added a test to make sure we got it all right.